mod_test/controllers.py

```diff
@@ -283,3 +284,24 @@ def download_build_log_file(test_id):
         raise TestNotFoundException('Build log for Test {id} not found'.format(id=test_id))
 
     raise TestNotFoundException('Test with id {id} not found'.format(id=test_id))
+
+
+# Define a web page to be accessed
+@mod_test.route('/svg/latest/')
+def svgbadge():
+    dwg = svgwrite.Drawing('status.svg', profile='full', size=('66px','18px')) #  dimensions from circle ci
+    # Get the latest Test
+    commit = Test.query.filter(Test.test_type != TestType.pull_request).first()
+    # Check it's status
+    if commit.TestStatus == TestStatus.completed:
+        dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+        dwg.save()
+    elif commit.TestStatus == TestStatus.canceled:
+        dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
+        dwg.save()
+    else:
+        dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
+        dwg.save()
+    response = make_response(open("status.svg").read())
```

Comment:
So you write an SVG to the disk, read it out again EACH time a request to the page gets made? That's horribly inefficient.